### ISSUE TITLE
LLVM 13.0.1 build by cmake failed

### ISSUE THREAD

**Issue by wangzy0327** [llvm/llvm-project#53812]
Version: 13.0.1
OS: Ubuntu 20.04
build generator: cmake

source file cmake build command line:
cmake  -DLLVM_ENABLE_PROJECTS="clang" -DLLVM_ENABLE_RUNTIMES="libcxx;libcxxabi" -DCMAKE_INSTALL_PREFIX=/home/wzy/llvm_workspace -DCMAKE_BUILD_TYPE=Debug -G "Unix Makefiles" ../llvm

then execute command line : make -j
the outputs display as below:
![image](https://user-images.githubusercontent.com/22990858/153816945-1a4bb14f-8736-4c35-80c0-02d2210c5f09.png)

how to solve it?

**Comment by tru** [comment]
You are running out of memory would be my guess. Try to limit the number of jobs instead of allowing make to spawn unlimited number of jobs: `make -j 6` or some other number depending on the number of cores and memory you have.

**Comment by wangzy0327** [comment]
OK, I will try it following your advice. Thanks!

**Comment by wangzy0327** [comment]
> You are running out of memory would be my guess. Try to limit the number of jobs instead of allowing make to spawn unlimited number of jobs: `make -j 6` or some other number depending on the number of cores and memory you have.

My cpu cores is 32. 


I try to limit the numbers of jobs :`make -j 20` and get outputs as below：
![image](https://user-images.githubusercontent.com/22990858/153823177-52ce96dd-f728-4cfc-9ae6-0a6b2c529653.png)

is it also too many numbers of jobs to make ?

**Comment by DimitryAndric** [comment]
Yes, now it is the linker that is out of memory. The link step can often require much more memory than compiling individual cpp files.

You can try adding `-D LLVM_PARALLEL_LINK_JOBS=4` to your CMake invocation, which will limit the number of parallel link jobs to 4. If that still leads to out-of-memory errors, lower the number even further.

**Comment by tru** [comment]
You can also try the following things:

* Switch your linker to gold: `-DLLVM_USE_LINKER=gold`
* Don't build a debug version of LLVM (unless you need it): `-DCMAKE_BUILD_TYPE=Release` 

gold and lld linkers have better memory management than the standard bfd linker. And without debug info the memory usage will definitely drop.
